components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -979,6 +1012,115 @@ private static void buildBasicInformation(IdentityProvider fedIdp, Map<String, S
         }
     }
 
+    private static String handleDeletion(String oldCertificateFile, String deletedCertificateValues) {
+
+        String decodedOldCertificate = new String(Base64.decode(oldCertificateFile));
+        String decodedDeletedCertificate = new String(Base64.decode(deletedCertificateValues));
+        boolean isDeleted;
+
+        int numberOfOldCertificates = StringUtils.countMatches(decodedOldCertificate, IdentityUtil.
+                PEM_BEGIN_CERTFICATE);
+        if (numberOfOldCertificates > 1) {
+            if (log.isDebugEnabled()) {
+                log.debug("More than one existing certificates found.");
+            }
+        }
+        StringBuilder latestCertificateValue = new StringBuilder();
+
+        // Iterate through existing certificates.
+        for (int i = 1; i <= numberOfOldCertificates; i++) {
```

Comment:
Here also I see some duplicate code from some other class. check whether we can move to some common place and reuse